fail.org exists and is for sale

Previously had an SSL error, now exists which is somewhat problematic. We should use https://httpstat.us/ or similar.

> See this failure: https://github.com/scikit-hep/pyhf/runs/2571497115?check_suite_focus=true

I'm confused though. This is a `ConnectionResetError` that doesn't seem to have anything to do with the code and doesn't show up again anywhere else in the runs on PR #1450

```pytb
>       assert (
            "SSLCertificateError" and "hostname 'www.fail.org' doesn't match" in ret.stderr
        )
E       assert ('SSLCertificateError' and "hostname 'www.fail.org' doesn't match" in 'Traceback (most recent call last):\n  File "/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/urllib...uests.exceptions.ConnectionError: (\'Connection aborted.\', ConnectionResetError(104, \'Connection reset by peer\'))\n')
E        +  where 'Traceback (most recent call last):\n  File "/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/urllib...uests.exceptions.ConnectionError: (\'Connection aborted.\', ConnectionResetError(104, \'Connection reset by peer\'))\n' = <pytest_console_scripts.RunResult object at 0x7fae1ed21040>.stderr
```

Can you explain in like very basic terms why this needs to get changed? I'm missing how we're doing anything wrong.

--- a/tests/test_scripts.py
+++ b/tests/test_scripts.py
@@ -560,7 +560,7 @@ def test_patchset_download(tmpdir, script_runner, archive):
         "pyhf.exceptions.InvalidArchiveHost: www.pyhfthisdoesnotexist.org is not an approved archive host"
         in ret.stderr
     )
-    # Valid URL but doesn't contain file
+    # Force a download from a real URL, but one that doesn't have an existing file
     command = f'pyhf contrib download --verbose --force https://httpstat.us/404/record/resource/1234567 {tmpdir.join("likelihoods").strpath}'
     ret = script_runner.run(*shlex.split(command))
     assert not ret.success

